src/concurrency/thread/crossbeam-spawn.md

```diff
@@ -28,13 +28,13 @@ fn find_max(arr: &[i32], start: usize, end: usize) -> i32 {
 
     let mid = start + (end - start) / 2;
     crossbeam::thread::scope(|scope| {
-        let left = scope.spawn(|| find_max(arr, start, mid));
-        let right = scope.spawn(|| find_max(arr, mid, end));
+        let left = scope.spawn(|_| find_max(arr, start, mid));
+        let right = scope.spawn(|_| find_max(arr, mid, end));
 
         // NOTE(unwrap): `join` will return an error if the thread panicked.
         // This way, panics will be propagated up to the `scope` call
         cmp::max(left.join().unwrap(), right.join().unwrap())
-    })
+    }).unwrap()
```

Comment:
Can you catch the thread panics rather than unwrapping them?

Reply:
What are you suggesting exactly?

We could call `.expect("a child thread panicked")` to display a more informative message. Or maybe print the panic errors and continue with the program. But it's better to just `.unwrap()` since we do the same thing one line above - we just propagate errors up.

In the end, this `.unwrap()` has no significance because we don't join any threads at the end of the scope - note that all spawned threads are already joined inside the scope.

Reply:
In the cookbook we try to reinforce good habits.  Even if the use case isn't as practical as another, we like to treach how to propogate errors up, like you say.  Here, the return type of the function is `i32` and if you instead return `Result<i32>`.

Reply:
I've updated the code so that `find_max()` returns an `Option<T>` (returns `None` when the array is empty). Does it look better now?